### DISCUSSION TITLE
How to open dialog only when request is taking too long time?

### DISCUSSION THREAD

**Discussion by sollyow** [cfug/dio#2197] [Q&A] [answered]
I want to implement simple dialog which should only show when request is taking more than 200ms.

**Comment by SoroushBeigi** [comment, marked as answer]
You can use a bool to know if the request is taking too long. it can be implemented like this:
```
//replace 200ms with desired timeout time
final timer = Timer(Duration(milliseconds: 200), () => emit(state.copyWith(isLoadingLong: true)));
await yourRequest();
timer.cancel();
emit(state.copyWith(isLoadingLong: false));
```
You can also use Dio's built-it CONNECT_TIMEOUT exception and set the timeout in BaseOptions
